doc/getting_started/build_sw.md

```diff
@@ -1,217 +1,335 @@
 `BUILD` files provide the information Bazel needs to build the targets in a directory.
 `BUILD` files also manage any subdirectories that don't have their own `BUILD` files.
 
-OpenTitan uses .bzl files to specify rules to build artifacts that require specific attention like target specific test rules and project specific binaries.
+OpenTitan uses .bzl files to specify custom rules to build artifacts that require specific attention like on-device test rules and project specific binaries.
 
-#### WORKSPACE file
+### WORKSPACE file
 
 The `WORKSPACE` file controls external dependencies such that builds can be made reproducible and hermetic.
-Bazel loads specific commits of external git repositories.
+Bazel loads specific external dependencies, such as various language toolchains.
 It uses them to build OpenTitan targets (like it does with bazel\_embedded) or to satisfy dependencies (as it does with abseil).
-To produce increasingly stable releases the `WORKSPACE` file will fix a number of these references to specific commits and not the default branch head.
-Dependencies will be added to the workspace so builds and tests are less sensitive to external updates and have a simpler getting started flow.
+To produce increasingly stable releases the external dependencies loaded in `WORKSPACE` file attempts to fix a all external `http_archive`s to a specific SHA.
+As we add more dependencies to the workspace, builds and tests will become less sensitive to external updates, and we will vastly simplify the [Getting Started]({{< relref "getting_started" >}}) instructions.
+
+### BUILD files
+
+Throughout the OpenTitan repository, `BUILD` files describe targets and dependencies in the same directory (and subdirectories that lack their own `BUILD` files).
+`BUILD` files are mostly hand-written.
+To maintain the invariant that hand-written files not be included in autogen directories, there are `BUILD` files that describe how to build and depend on auto-generated files in autogen subdirectories.
+
+## Linting Software
 
-#### BUILD files
+There are several Bazel rules that enable running quality checks and fixers on code.
+The subsections below describe how to use them.
+All of the tools described below are run in CI on every pull request, so it is best to run them before committing code.
 
-Throughout the OpenTitan repo, `BUILD` files describe targets and dependencies in the same directory and subdirectories that lack their own `BUILD` files.
-`BUILD` files are mostly hand-written; to suit the request that hand-written files not be included in autogen directories, there are large `BUILD` files that describe how to build and depend on auto-generated files in autogen subdirectories.
+### Linting C/C++ Code
+The OpenTitan supported linter for C/C++ files is `clang-format`.
+It can be run with Bazel as shown below.
+
+Run the following to check if all C/C++ code as been formatted correctly:
```

Comment:
dumb question.. do we still have the issue of mismatched clang-format with CI in this case..?

Reply:
So I think Bazel is using the [one provided with the lowrisc tools chain](https://github.com/lowRISC/opentitan/blob/058344a4fc2a27460060718315407f4675fd5f00/rules/quality.bzl#L101), which is what we want, but I think CI [still uses the system installed one](https://github.com/lowRISC/opentitan/blob/058344a4fc2a27460060718315407f4675fd5f00/ci/scripts/clang-format.sh#L33) (version 6.0).

So CI should probably be updated to use the bazel managed version. I filed issue #13163 to track.